src/test/kotlin/com/example/flashcardbackend/deck/DeckControllerTest.kt

```diff
@@ -0,0 +1,186 @@
+package com.example.flashcardbackend.deck
+
+import com.example.flashcardbackend.flashcard.FlashcardListItem
+import org.hamcrest.Matchers
+import org.junit.jupiter.api.Test
+import org.mockito.Mockito.*
+import org.springframework.beans.factory.annotation.Autowired
+import org.springframework.boot.test.autoconfigure.web.servlet.WebMvcTest
+import org.springframework.boot.test.mock.mockito.MockBean
+import org.springframework.http.MediaType
+import org.springframework.test.web.servlet.MockMvc
+import org.springframework.test.web.servlet.request.MockMvcRequestBuilders.*
+import org.springframework.test.web.servlet.result.MockMvcResultMatchers.*
+
+@WebMvcTest(DeckController::class, DeckService::class)
```

Comment:
This works, but I like use the standalone configuration because if I have to configure everything manually, it's pretty obvious if my code has too many dependencies. I even have a blog post about this: https://www.petrikainulainen.net/programming/testing/the-best-way-to-configure-the-spring-mvc-test-framework-part-one/